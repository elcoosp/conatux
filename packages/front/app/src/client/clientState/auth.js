import { STORAGE } from '@:/constants'

export default {
  token: localStorage.getItem(STORAGE.KEYS.TOKEN)
}
