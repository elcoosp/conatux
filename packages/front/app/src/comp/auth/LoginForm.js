import React from 'react'
import { Mutation, ApolloConsumer } from 'react-apollo'

import { Fields, Form, validation } from '@:/comp/helpers/form'
import Auth from '@:/services/auth'
import { LOGIN } from '@:/client/mutations'

const validationSchema = validation.makeSchemaFor('email', 'password')

export const makeSubmitHandler = ({ client, login }) => async (
  variables,
  { setSubmitting, setErrors } // Formik handler functions
) => {
  try {
    const {
      data: {
        login: { token }
      }
    } = await login({ variables })
    Auth.storeToken(token)
    client.writeData({ data: { auth: { token } } })
  } finally {
    setSubmitting(false)
  }
}

const LoginForm = () => (
  <ApolloConsumer>
    {client => (
      <Mutation mutation={LOGIN}>
        {(login, { loading, error, data }) => (
          <Form
            validationSchema={validationSchema}
            onSubmit={makeSubmitHandler({ client, login })}
            apolloError={error}
          >
            <Fields.Email />
            <Fields.Password />
          </Form>
        )}
      </Mutation>
    )}
  </ApolloConsumer>
)

export default LoginForm
