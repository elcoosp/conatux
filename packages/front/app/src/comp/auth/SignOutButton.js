import React from 'react'

import Auth from '@:/services/auth'
import Buttons from 'ui/buttons'

const SignOutButton = () => (
  <Buttons.Basic onClick={Auth.signOut}>Sign out</Buttons.Basic>
)
export default SignOutButton
