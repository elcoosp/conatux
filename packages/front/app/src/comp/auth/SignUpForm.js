import React from 'react'
import * as yup from 'yup'
import { LabelledField, Form } from '@:/comp/helpers/form'

// Use spected
const validationSchema = yup.object().shape({
  username: yup
    .string()
    .required()
    .min(4)
    .max(12),
  email: yup
    .string()
    .email()
    .max(40)
    .required()
})

const initialValues = {
  email: '',
  password: '',
  passwordConfirmation: '',
  username: ''
}

const SignUpForm = () => (
  <Form
    initialValues={initialValues}
    validationSchema={validationSchema}
    // onSubmit={(values, { setSubmitting }) => {}}
  >
    <LabelledField label="Username" type="text" name="username" />
    <LabelledField label="Email" type="email" name="email" />
    <LabelledField label="Password" type="password" name="password" />
    <LabelledField
      label="Password confirmation"
      type="password"
      name="passwordConfirmation"
    />
  </Form>
)

export default SignUpForm
