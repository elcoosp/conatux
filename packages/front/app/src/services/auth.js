import { STORAGE } from '@:/constants'
import client from '@:/client'
import cache, { persistor } from '@:/client/cache'
const clearApolloCache = () =>
  localStorage.removeItem(STORAGE.KEYS.APOLLO_CACHE)
const getStoredToken = () => localStorage.getItem(STORAGE.KEYS.TOKEN)
const storeToken = token => localStorage.setItem(STORAGE.KEYS.TOKEN, token)
const clearStoredToken = () => localStorage.removeItem(STORAGE.KEYS.TOKEN)

const signOut = async () => {
  try {
    auth.clearStoredToken()
    await client.resetStore() // Clear the in memory cache
  } catch (error) {
    /* Shallow this error if it's about request in flight since easiest way to really have a cean this.state.
     See https://github.com/apollographql/apollo-client/issues/2919
     */
    const resetStoreWhileInFlightQueriesPendingMessage =
      'Network error: Store reset while query was in flight(not completed in link chain)'
    if (error.message !== resetStoreWhileInFlightQueriesPendingMessage) {
      throw error
    }
  }
}
const auth = {
  getStoredToken,
  storeToken,
  clearStoredToken,
  clearApolloCache,
  signOut
}

export default auth
