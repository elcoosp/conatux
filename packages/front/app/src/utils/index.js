export { default as testing } from './testing'
export { default as linkify } from './linkify'
