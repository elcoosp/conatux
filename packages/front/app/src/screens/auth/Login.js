import React from 'react'
import { Redirect } from 'react-router-dom'
import P from 'prop-types'

import { LoginForm } from '@:/comp/auth'
import { Screen } from '@:/screens'
import { ROUTES } from '@:/constants'
import { withAuthToken } from '@:/comp/helpers/hoc'

export const Login = ({ authToken }) =>
  authToken ? (
    <Redirect to={{ pathname: ROUTES.Me.Profile.path }} />
  ) : (
    <Screen title={ROUTES.Auth.Login.label}>
      <LoginForm />
    </Screen>
  )

Login.propTypes = {
  authToken: P.string
}

export default withAuthToken(Login)
