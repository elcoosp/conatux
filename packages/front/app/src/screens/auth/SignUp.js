import React from 'react'
import { Screen, Links } from '@:/screens'
import { ROUTES } from '@:/constants'
import { SignUpForm } from '@:/comp/auth'

const SignUp = () => (
  <Screen title={ROUTES.Auth.SignUp.label}>
    <Links.Me.Profile />
    <SignUpForm />
  </Screen>
)

export default SignUp
