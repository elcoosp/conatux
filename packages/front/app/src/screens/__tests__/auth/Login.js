import { Login } from '../../auth/Login'
import { desc } from '@:/utils/testing'

desc.compWithSnap({ Login }, { authToken: 'validToken' }, () => {})
desc.compWithSnap({ Login }, { authToken: null }, () => {})
