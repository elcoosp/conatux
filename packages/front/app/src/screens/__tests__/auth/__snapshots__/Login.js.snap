// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/me/profile",
    }
  }
/>
`;

exports[`<Login /> should match snapshot 2`] = `
<Screen
  title="Login"
>
  <LoginForm />
</Screen>
`;
